modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/DynamicInlineScroll.es.js

```diff
@@ -148,12 +148,13 @@ class DynamicInlineScroll extends PortletBase {
 	onScroll_(event) {
 		const {cur, initialPages, pages} = this;
 		const {target} = event;
-
-		let pageIndex = Number(target.dataset.pageIndex);
-		let pageIndexMax = Number(target.dataset.maxIndex);
+		let pageIndex = Number(target.getAttribute('data-page-index'));
```

Comment:
I think you can create getNumber again and then use [Marko's suggestion](https://github.com/liferay-frontend/liferay-portal/pull/2523#issuecomment-1190013072)